scripts/install-windows-prereqs.ps1

```diff
@@ -588,28 +589,82 @@ function Install-Docker {
     Write-Output '[WARNING] Docker is no longer a prerequisite and installation thereof is deprecated.'
 }
 
-try {
+function Validate-SkipTasks {
+    Param(
+        [Parameter(Mandatory=$true)][array]$SkipTaskLower
+    )
+    $ValidSet = @("Nuget3", "Nuget6", "Python", "VisualStudio", "LLVM", "Git", "Shellcheck", "NSIS", "Docker", "OpenSSL", "VCRuntime")
+    foreach($item in $SkipTaskLower) {
+        if($ValidSet -notcontains $item) {
+            Write-Output "$item is not a valid package to skip."
+            return
+        }
+        Write-Output "Skipping $PackageToInstall"
+        return
+    }
+}
 
+try {
     if ($VerificationOnly) {
         Write-Output 'Verification only - Complete'
         Exit 0
     }
-
+    $lowercaseSkipTask = $SkipTask.ForEach({$_.ToLower()})
+    Validate-SkipTasks -SkipTaskLower $lowercaseSkipTask
     Start-LocalPackagesDownload
-
     Install-7Zip
-    Install-Nuget -Version 3
-    Install-Nuget -Version 6
-    Install-Python3
-    if (!$SkipVSInstall) {
+    if($lowercaseSkipTask -contains "nuget3") {
+        Write-Output "Skipping Nuget3"
+    }
+    else {
+        Install-Nuget -Version 3
+    }
+    if($lowercaseSkipTask -contains "nuget6") {
+        Write-Output "Skipping Nuget6"
+    }
+    else {
+        Install-Nuget -Version 6
+    }
+    if($lowercaseSkipTask -contains "python") {
+        Write-Output "Skipping Python"
+    }
+    else {
+        Install-Python3
+    }
+    if ($lowercaseSkipTask -contains "visualstudio" -or $SkipVSInstall) {
+        Write-Output "Skipping VisualStudio"
+    }
+    else{
         Install-VisualStudio
     }
-    Install-LLVM
-    Install-Git
-    Install-Shellcheck
-    Install-NSIS
-
-    if ($InstallDocker) {
+    if ($lowercaseSkipTask -contains "llvm") {
+        Write-Output "Skipping LLVM"
+    }
+    else {
+        Install-LLVM
+    }
+    if ($lowercaseSkipTask -contains "git") {
+        Write-Output "Skipping Git"
+    }
+    else {
+        Install-Git
+    }
+    if ($lowercaseSkipTask -contains "shellcheck") {
+        Write-Output "Skipping Shellcheck"
+    }
+    else {
+        Install-Shellcheck
+    }
+    if ($lowercaseSkipTask -contains "nsis") {
+        Write-Output "Skipping NSIS"
+    }
+    else {
+        Install-NSIS
+    }
+    if($lowercaseSkipTask -contains "docker" -or !$InstallDocker) {
```

Comment:
Requested change: Docker is skipped by default, and even if it isn't skipped we don't install it. So this doesn't need to be added and all Docker related code can be removed entirely.